### PR TITLE
APMrover2 : correct arming and failsafe in skid steer mode

### PR DESCRIPTION
**Pull request by khancyr**
Commit https://github.com/diydrones/ardupilot/commit/3636b5331310b417db38c467154f05929ff250f3#diff-e22a85a55f71f1b9b2d3f293dea61368 introduce arming for rover. But in skid steering mode it was only applied on throttle and the rover still pivot.
The patch also correct rover behaviour in case of failsafe or loitering.

### CODE REVIEW

**Comment by khancyr** [comment]
It has been test on a tank style rover only. My rover use an arduino to translate RC-PWM from Pixhawk to H-Bridges. So another test could be useful.
It must be test on non-skidsteer rover !

**Comment by gmorph** [comment]
Thanks for the PR khancyr.  My skid steering rover is in bits at the moment but I will put it back together so I can test this.  I should be able to test it on a non skid steering rover easily enough in the mean time.  Thanks!

**Comment by rmackay9** [comment]
I think we've got a newer PR for this issue so closing this one.
https://github.com/ArduPilot/ardupilot/pull/3828

**Comment by khancyr** [comment]
@rmackay9 ,
Hi,
This commit is not the same as #3828 . In this one I correct only skid steer action, and it is working.
On the other one, I deal with radio, but some bug aren't solved.
 I can merge the two in one PR if need. But I am not sure about the radio PR.

**Comment by rmackay9** [comment]
Ah, ok, sorry.  re-opened.

**Comment by AndWik** [comment]
As the motors on one side engage without arming must be considered a bug!
Implementing Arming is a good feature, I think, but as the rover will go circles only, until armed, isn't correct according to me.

Thanks
Anders

**Comment by khancyr** [comment]
@gmorph, Could you test my PR on you skid steer rover. Normally it works but I want a second testing on a real skid steer setting ( I use arduino to translate RC pwm to PWM )
This will solve http://discuss.ardupilot.org/t/robot-spins-in-circle-in-steering-mode-rover-3-0/9067/2 and http://discuss.ardupilot.org/t/skid-steering-isnt-working/8940/3 behaviour .

I try to reproduce this bug on stil but it is not present ... ( I didn't dig far in rover skid steer model to find why)

When I got more time, I will look for other bug

File: APMrover2/Steering.cpp [hunk 1]
```diff
@@ -239,10 +247,18 @@ void Rover::set_servos(void) {
         if ((failsafe.bits & FAILSAFE_EVENT_THROTTLE) && control_mode < AUTO) {
             // suppress throttle if in failsafe
             channel_throttle->servo_out = 0;
+            // suppress steer if in failsafe and skid steer mode
+            if (g.skid_steer_out) {
+                channel_steer->radio_out = 0;
```
Comment (gmorph) [inline APMrover2/Steering.cpp]: This should be channel_steer->servo_out = 0;
Can you update please?

File: APMrover2/Steering.cpp [hunk 2]
```diff
@@ -239,10 +247,18 @@ void Rover::set_servos(void) {
...
+            if (g.skid_steer_out) {
+                channel_steer->radio_out = 0;        
+            }
         }
 
         if (!hal.util->get_soft_armed()) {
             channel_throttle->servo_out = 0;
+            // suppress steer if in failsafe and skid steer mode
+            if (g.skid_steer_out) {
+                channel_steer->radio_out = 0;
```
Comment (gmorph) [inline APMrover2/Steering.cpp]: This should be channel_steer->servo_out = 0;
Can you update please?

File: APMrover2/Steering.cpp [hunk 3]
```diff
@@ -86,6 +86,10 @@ void Rover::calc_throttle(float target_speed) {
     // then set the throttle to minimum
     if (!auto_check_trigger() || ((loiter_time > 0) && (control_mode == AUTO))) {
         channel_throttle->servo_out = g.throttle_min.get();
+        // Stop rotation in case of loitering and skid steering
+        if (g.skid_steer_out) {
+            channel_steer->radio_out = 0;
```
Comment (gmorph) [inline APMrover2/Steering.cpp]: This should be channel_steer->servo_out = 0;
Can you update please?

**Comment by gmorph** [comment]
khancyr - once you make those changes above I will re-test and merge this in.
Thanks, Grant.

**Comment by khancyr** [comment]
@gmorph I rebase on master , made correction you pointed and update to new servo syntaxe !
I success to see the bug on sitl, just need to use rover-skid, go to guided without arm and set a target position. The rover will turn... I can't see loitering bug on sitl ...
Now it should work.

**Comment by gmorph** [comment]
Yes I submitted a new PR with the rover-skid all setup and ready to go so it should make SITL easier for everyone with a skid steering rover.  
I appreciate you making the changes.  I'll test and see how we go.
Thanks, Grant.

File: APMrover2/Steering.cpp [hunk 4]
```diff
@@ -223,6 +227,10 @@ void Rover::set_servos(void) {
         if (failsafe.bits & FAILSAFE_EVENT_THROTTLE) {
             // suppress throttle if in failsafe and manual
             channel_throttle->set_radio_out(channel_throttle->get_radio_trim());
+            // suppress steer if in failsafe and manual and skid steer mode
+            if (g.skid_steer_out) {
+                channel_steer->set_servo_out(channel_steer->get_radio_trim());
```
Comment (gmorph) [inline APMrover2/Steering.cpp]: Still not right.  This should be 
channel_steer->set_radio_out(channel_steer->get_radio_trim());
You can't mix the servo and radio settings.

**Comment by khancyr** [comment]
It should be good ! Sorry for the mistake ...

**Comment by gmorph** [comment]
Applied - thanks!

**Pull request closed without merging**